frontend/src/routes/GymPage/index.tsx

```diff
@@ -49,22 +57,77 @@ export default function GymPage({ onLeftClick }: { onLeftClick?: () => void }) {
       userID: 'A0234567M',
     },
   ]
+  useEffect(() => {
+    dispatch(getGymStatus())
+    dispatch(getUserDetail())
+    // dispatch(getGymHistory())
+  }, [])
 
   const [currentTab, setCurrentTab] = useState<number>(1)
   const sections = ['Gym', 'History']
   const history = useHistory()
+  //TODO: export this
+  const buttonText = ['Key With Me', 'Return Key', 'Close Gym', 'Open Gym']
```

Comment:
maybe can represent in a map, so to access you don't need index numbers to get them, which is another area for errors.

actionButtonLabel.keyWithMe: 'Key With Me'
actionButtonLabel.returnKey: 'Return Key'